src/QmlControls/QGCButton.qml

```diff
@@ -11,32 +12,50 @@ Button {
 
     property var __qgcPal: QGCPalette { colorGroupEnabled: enabled }
 
-    property bool __showHighlight: pressed | hovered | checked
+    property bool __showHighlight: (pressed | hovered | checked) && !__forceHoverOff
 
-    style: ButtonStyle {
-/*
-            background: Rectangle {
-                implicitWidth: 100
-                implicitHeight: 25
-                color: __showHighlight ?
-                    control.__qgcPal.buttonHighlight :
-                    (primary ? control.__qgcPal.primaryButton : control.__qgcPal.button)
-            }
+    // This fixes the issue with button hover where if a Button is near the edge oa QQuickWidget you can
+    // move the mouse fast enough such that the MouseArea does not trigger an onExited. This is turn
+    // cause the hover property to not be cleared correctly.
 
-            label: Text {
-                width: parent.width
-                height: parent.height
+    property bool __forceHoverOff: false
 
-                verticalAlignment: TextEdit.AlignVCenter
-                horizontalAlignment: TextEdit.AlignHCenter
+    property int __lastMouseX:  __behavior.mouseX
+    property int __lastMouseY:  __behavior.mouseY
+    property int __lastCursorX: 0
+    property int __lastCursorY: 0
 
-                text: control.text
-                color: __showHighlight ?
-                    control.__qgcPal.buttonHighlightText :
-                    (primary ? control.__qgcPal.primaryButtonText : control.__qgcPal.buttonText)
+    property MousePosition __mousePosition: MousePosition { }
+
+    Connections {
+        target: __behavior
+        onMouseXChanged: {
+            __lastCursorX = __mousePosition.mouseX
+            __lastCursorY = __mousePosition.mouseY
+        }
+        onMouseYChanged: {
+            __lastCursorX = __mousePosition.mouseX
+            __lastCursorY = __mousePosition.mouseY
+        }
+        onEntered: { __forceHoverOff; false; hoverTimer.start() }
+        onExited: { __forceHoverOff; false; hoverTimer.stop() }
+    }
+
+    Timer {
+        id:         hoverTimer
+        interval:   250
+        repeat:     true
+
+        onTriggered: {
+            if (__lastCursorX != __mousePosition.mouseX || __lastCursorY != __mousePosition.mouseY) {
+                __forceHoverOff = true
+            } else {
+                __forceHoverOff = false
             }
```

Comment:
As far as I can tell, you cannot get a world coordinate from within QML. What are available are methods for mapping coordinates between elements (Items) and their parents/children. Never outside. The coordinates you get from mouseX/Y within the MouseArea are based on its bounding rectangle (0,0 to width,height). The coordinates you get from QPoint::pos() are global coordinates (the entire screen). To compare mouseX to QPoint::pos().x you would need to map from the Item's coordinate space to world space. The best I can think of is passing the QObject as an argument to the c++ side and map it there.